store/tx_manager.go

```diff
@@ -330,19 +337,43 @@ func (txm *EthTxManager) ContractLINKBalance(wr models.WithdrawalRequest) (asset
 	return *linkBalance, nil
 }
 
+// newContractsBox returns the packr.Box instance that holds the contract JSON files
+func newContractsBox() packr.Box {
+	return packr.NewBox("../solidity/build/contracts")
+}
+
+// Transact converts arguments for a method in a contract to a byte array that can be used to interact with the contract
+func (txm *EthTxManager) Transact(contractName string, method string, args ...interface{}) ([]byte, error) {
+	jsonFile, err := txm.contracts.MustBytes(contractName + ".json")
+	if err != nil {
+		return nil, errors.New("unable to read contract JSON")
+	}
+
+	var contract struct {
+		ABI interface{} `json:"abi"`
+	}
+	err = json.Unmarshal(jsonFile, &contract)
+	if err != nil {
+		return nil, err
+	}
+
+	abiBytes, err := json.Marshal(contract.ABI)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I would break this up into an object. Pull everything in this method out into it's the initialization method, and then you can have a method for encoding arguments. Breaking it up like this makes it a bit more composable and easier to test.

For example, instead of the place where you call `Transact` below:
```go
oracle, err := GetContract("Oracle")
data, err := oracle.encodeMessageCall("withdraw", recipient, amount)
```